main.cpp

```diff
@@ -71,8 +71,12 @@ struct CompareF                                //4
 {
     T* compare(T* a, T* b) //5
     {
-        if( a->value < b->value ) return a;
-        if( a->value > b->value ) return b;
+        if (a&&b !=nullptr)
```

Comment:
What you wrote equates to this logic:
```
if( a is not zero, and b is specifically not nullptr )
```
- [ ] revise with correct checks for nullptr.